services/autoscaling/tests/unit/test_utils_docker.py

```diff
@@ -397,14 +398,35 @@ async def test_pending_service_task_with_insufficient_resources_properly_sorts_t
         days=1
     )
     for task in pending_tasks:
-        assert task.CreatedAt
+        assert task.CreatedAt  # NOTE: in this case they are but they might be None
         assert (
             to_datetime(task.CreatedAt).replace(tzinfo=datetime.timezone.utc)
             > last_date
         )
         last_date = to_datetime(task.CreatedAt).replace(tzinfo=datetime.timezone.utc)
 
 
+def test_safe_sort_key_callback():
+    tasks_with_faulty_timestamp = [
+        Task(ID=n, CreatedAt=value)  # type: ignore
+        for n, value in enumerate(
+            [
+                # SEE test_to_datetime_conversion_known_errors
+                None,
+                "2023-03-15 09:20:58.123456",
+                "2023-03-15T09:20:58.123456",
+                "2023-03-15T09:20:58.123456Z",
```

Comment:
sometimes it also has more numbers before the Z right?

Reply:
yes